nebula-logger/core/main/logger-engine/classes/ComponentLogger.cls

```diff
@@ -76,73 +79,75 @@ public inherited sharing class ComponentLogger {
         }
     }
 
-    private static void setBrowserDetails(LogEntryEventBuilder logEntryEventBuilder, ComponentLogEntry componentLogEntry) {
-        logEntryEventBuilder.getLogEntryEvent().BrowserFormFactor__c = componentLogEntry.browserFormFactor;
-        logEntryEventBuilder.getLogEntryEvent().BrowserLanguage__c = componentLogEntry.browserLanguage;
-        logEntryEventBuilder.getLogEntryEvent().BrowserScreenResolution__c = componentLogEntry.browserScreenResolution;
-        logEntryEventBuilder.getLogEntryEvent().BrowserUrl__c = componentLogEntry.browserUrl;
-        logEntryEventBuilder.getLogEntryEvent().BrowserUserAgent__c = componentLogEntry.browserUserAgent;
-        logEntryEventBuilder.getLogEntryEvent().BrowserWindowResolution__c = componentLogEntry.browserWindowResolution;
+    private static void setBrowserDetails(LogEntryEvent__e logEntryEvent, ComponentLogEntry componentLogEntry) {
+        logEntryEvent.BrowserFormFactor__c = componentLogEntry.browserFormFactor;
+        logEntryEvent.BrowserLanguage__c = componentLogEntry.browserLanguage;
+        logEntryEvent.BrowserScreenResolution__c = componentLogEntry.browserScreenResolution;
+        logEntryEvent.BrowserUrl__c = componentLogEntry.browserUrl;
+        logEntryEvent.BrowserUserAgent__c = componentLogEntry.browserUserAgent;
+        logEntryEvent.BrowserWindowResolution__c = componentLogEntry.browserWindowResolution;
     }
 
-    private static void setComponentErrorDetails(LogEntryEventBuilder logEntryEventBuilder, ComponentError componentError) {
+    private static void setComponentErrorDetails(LogEntryEvent__e logEntryEvent, ComponentError componentError) {
         if (componentError == null) {
             return;
         }
 
-        logEntryEventBuilder.getLogEntryEvent().ExceptionMessage__c = componentError.message;
-        logEntryEventBuilder.getLogEntryEvent().ExceptionStackTrace__c = componentError.stack;
-        logEntryEventBuilder.getLogEntryEvent().ExceptionType__c = componentError.type;
+        logEntryEvent.ExceptionMessage__c = componentError.message;
+        logEntryEvent.ExceptionType__c = componentError.type;
+
+        LoggerStackTrace.SourceLanguage language = LoggerStackTrace.SourceLanguage.JavaScript;
+        // Each of the built-in exceptions use the fully-qualified name (e.g., System.DmlException)
+        // FIXME this probably doesn't work with custom exception classes - additional changes
```

Comment:
don't _all_ exception classes have to end with `Exception` in order to be valid?

Reply:
Oooooh, right! Yeah, I'll switch to using `endsWith('Exception')` instead